portal/utils/updater.js

```diff
@@ -0,0 +1,29 @@
+const { exec } = require("child_process");
+
+const {createHmac, timingSafeEqual} = require("node:crypto");
+
+const logger = require("./logger")
+
+function compare_signature(secret, signature, payload){
+	const hmac = createHmac('sha256', secret=secret)
+	hmac.update(payload);
+	const new_signature = `sha256=${hmac.digest('hex')}`;
+	return (timingSafeEqual(Buffer.from(signature), Buffer.from(new_signature)));
+}
+
+function pull_and_restart(){
+	logger.info("Git Pulling...");
+	exec("git pull", (error, stdout, stderr)=>{
+		if(error){
+			logger.error(`Git Pull errored: ${error}`);
+		} else {
+			logger.info("Restarting using systemctl...");
+			exec("sudo systemctl restart casbot.target");
+		}
+	});
+}
```

Comment:
Please use arrow functions instead.

Reply:
Why so?

Reply:
I am actually curious.

Reply:
Interesting. This is a very unique style guide. But it makes sense 👍